### PR TITLE
Series label format and formatter

### PR DESCRIPTION
**Pull request by TorsteinHonsi**
Added new series label options, `series.label.format` and `series.label.formatter`.

The PR also fixes some issues related to the series-label module combined with the rest of the unit tests.

### CODE REVIEW

File: ts/modules/series-label.src.ts [hunk 1]
```diff
@@ -1032,25 +1067,23 @@ Chart.prototype.drawSeriesLabels = function (): void {
...
                         attr = {};
                     }
-                    (series.labelBySeries as any)
+
+                    // Default initial animation to a fraction of the series
+                    // animation (#9396)
+                    let animationOptions: Highcharts.AnimationOptionsObject|undefined;
+                    if (isNew) {
+                        animationOptions = animObject(series.options.animation);
+                        (animationOptions.duration as number) *= 0.2;
```
Comment (bre1470) [inline ts/modules/series-label.src.ts]: This kind of casting (`as number`) should be avoided, as it hides undefined/null problems.
- If duration is always set, declaration typing of the property should be changed instead.
- If optional, do `animationOptions.duration = (animationOptions.duration || 0) * 0.2` instead.
- If you can not deal with it now, cast it to `any` and we will pick it up later.

File: ts/modules/series-label.src.ts [hunk 2]
```diff
@@ -174,6 +177,28 @@ H.setOptions({
...
+                 * A format string for the label, with support for a subset of
+                 * HTML. Variables are enclosed by curly brackets. Available
+                 * variables are `name`, `options.xxx`, `color` and other
+                 * members from the `series` object. Use this option also to set
+                 * a static text for the label.
+                 *
+                 * @type string
+                 * @since next
+                 */
+                format: void 0,
```
Comment (pawelfus) [inline ts/modules/series-label.src.ts]: Should this be simply `format: {name}`?

Comment (pawelfus) [reply ts/modules/series-label.src.ts]: Ah, I see, then `formatter` will never be called.

File: ts/modules/series-label.src.ts (hunk 1 above)
Comment (pawelfus) [reply ts/modules/series-label.src.ts]: If (or when) #13005 will be merged, then `animObject()` will always return `{ duration, defer }` and casting to `any` will be unnecessary. So I would go for now with the third option :)

Comment (bre1470) [reply ts/modules/series-label.src.ts]: Fantastic! 👍

Comment (TorsteinHonsi) [reply ts/modules/series-label.src.ts]: Great!

**Comment by TorsteinHonsi** [comment]
> Where AnimationOptionsObject is used as an input, it should become Partial<AnimationOptionsObject>. What do you think?

Yes absolutely. This pattern is all over in the Highcharts source code. Would love to see it implemented.

**Pull request merged**